diesel_cli/src/infer_schema_internals/mysql.rs

```diff
@@ -173,3 +256,85 @@ fn unsigned_types_are_supported() {
 fn types_with_space_are_not_supported() {
     assert!(determine_type_name("lol wat").is_err());
 }
+
+#[cfg(test)]
+mod test {
+    extern crate dotenvy;
+
+    use self::dotenvy::dotenv;
+    use super::*;
+    use std::env;
+
+    fn connection() -> MysqlConnection {
+        dotenv().ok();
+
+        let connection_url = env::var("MYSQL_DATABASE_URL")
+            .or_else(|_| env::var("DATABASE_URL"))
+            .expect("DATABASE_URL must be set in order to run tests");
+        let mut connection = MysqlConnection::establish(&connection_url).unwrap();
+        connection.begin_test_transaction().unwrap();
+        connection
+    }
+    #[test]
+    #[ignore = "FIXME: Figure out how to handle tests that modify schema"]
+    fn get_table_data_loads_column_information() {
+        let mut connection = connection();
+
+        diesel::sql_query("CREATE SCHEMA test_schema")
+            .execute(&mut connection)
+            .unwrap();
+        diesel::sql_query(
+                "CREATE TABLE test_schema.table_1 (id SERIAL PRIMARY KEY COMMENT 'column comment') COMMENT 'table comment'",
+            ).execute(&mut connection)
+            .unwrap();
+        diesel::sql_query("CREATE TABLE test_schema.table_2 (id SERIAL PRIMARY KEY)")
+            .execute(&mut connection)
+            .unwrap();
+
+        let table_1 = TableName::new("table_1", "test_schema");
+        let table_2 = TableName::new("table_2", "test_schema");
+
+        let id_with_comment = ColumnInformation::new(
+            "id",
+            "bigint unsigned",
+            None,
+            false,
+            Some("column comment".to_string()),
+        );
+        let id_without_comment = ColumnInformation::new("id", "bigint unsigned", None, false, None);
+        assert_eq!(
+            Ok(vec![id_with_comment]),
+            get_table_data(&mut connection, &table_1, &ColumnSorting::OrdinalPosition)
+        );
+        assert_eq!(
+            Ok(vec![id_without_comment]),
+            get_table_data(&mut connection, &table_2, &ColumnSorting::OrdinalPosition)
+        );
+    }
+
+    #[test]
+    #[ignore = "FIXME: Figure out how to handle tests that modify schema"]
```

Comment:
This can be solved by using a temporary table: https://dev.mysql.com/doc/refman/8.0/en/create-temporary-table.html

Reply:
This doesn't seem to work unfortunately. It appears that info on temporary tables is not stored in `information_schema.tables` where the comment information is fetched from but in [information_schema.innodb_temp_table_info](https://dev.mysql.com/doc/mysql-infoschema-excerpt/8.0/en/information-schema-innodb-temp-table-info-table.html) which doesn't provide the comment info.

Would it be acceptable to do something like
```sql
DROP TABLE IF EXISTS diesel_test.get_table_data_loads_column_information; # named after the test to avoid conflicts with other tests
CREATE TABLE diesel_test.get_table_data_loads_column_information # ...
```
optionally with another drop table at the end as cleanup?

Reply:
Yes that would be also fine. There are other tests in diesel_cli that modify the schema as well so I do not see any reason for not doing that here as well.